.github/workflows/distribute-apps.yml

```diff
@@ -7,36 +7,41 @@ on:
         required: true
         type: string
         description: 'Either development or production'
-
-
-  workflow_dispatch:
-    inputs:
-      environment:
-        required: true
+      version_string:
+        required: false
         type: string
-        description: 'Either development or production'
+        description: 'The version string to be used for the app'
 
 jobs:
   distribute-ios:
     runs-on: macos-14
     environment: ${{ inputs.environment }}
     steps:
-      - uses: actions/checkout@v4
+      - name: Checkout Tag
+        if: inputs.environment == 'production'
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.version_string}}
+      - name: Checkout Branch
+        if: inputs.environment == 'development'
+        uses: actions/checkout@v4
       - uses: actions/setup-node@v4
         with:
           node-version-file: '.node-version'
           cache: yarn
       - uses: ruby/setup-ruby@v1
       - uses: maxim-lobanov/setup-xcode@v1
         with:
-          xcode-version: '15.2'
+          xcode-version: '16.0'
```

Comment:
Will this require everyone to be running 16+

Reply:
I don't believe it should